src/coreclr/jit/gentree.cpp

```diff
@@ -24855,9 +24940,10 @@ GenTree* Compiler::gtNewSimdShuffleNode(
     if (simdSize == 32)
     {
         assert(compIsaSupportedDebugOnly(InstructionSet_AVX2));
-
-        if ((varTypeIsByte(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512VBMI_VL)) ||
-            (varTypeIsShort(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512BW_VL)))
+        bool isAvx10v1 = compOpportunisticallyDependsOn(InstructionSet_AVX10v1);
+        if (((varTypeIsByte(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512VBMI_VL)) ||
+             (varTypeIsShort(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512BW_VL))) &&
+            (!isAvx10v1))
```

Comment:
`compOpportunisticallyDependsOn` flags a method as opportunistically using the ISA it checks and so it will require a standard user method to be rejitted in the case the underlying hardware supports the ISA.

As such, we should really only do checks in paths that we know will actually use the hardware intrinsics and so it'd be better to refactor the code so that the `compOpportunisticallyDependsOn(InstructionSet_AVX10v1)` is only done if `varTypeIsByte || varTypeIsShort`.

I imagine we may simply want a helper method here to make the `Is Avx10 || isa` check simpler

Reply:
I have made the change to following

```cpp
        if (((varTypeIsByte(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512VBMI_VL)) ||
             (varTypeIsShort(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512BW_VL))) &&
            ((varTypeIsByte(simdBaseType) || varTypeIsShort(simdBaseType)) && !isAvx10v1))
```

Are you suggesting we have a method to check for ```compIsAvx10OrIsa(isa)``` helper and use it as below?

```cpp
        if (((varTypeIsByte(simdBaseType) && !compIsAvx10OrIsa(InstructionSet_AVX512VBMI_VL)) ||
             (varTypeIsShort(simdBaseType) && !compIsAvx10OrIsa(InstructionSet_AVX512BW_VL))))
```

Reply:
> Are you suggesting we have a method to check for compIsAvx10OrIsa(isa) helper and use it as below?

Yes, I think it would be beneficial to have something like the `bool compIsEvexOpportunisticallySupported(bool* isV512Supported)` described above. We could make it account for this scenario too by having:
```cpp
bool compIsEvexOpportunisticallySupported(bool* isV512Supported, InstructionSet avx512InstructionSet = InstructionSet_AVX512F)
{
    // TODO: assert avx512InstructionSet is one of the AVX512 instruction sets

    if (compOpportunisticallyDependsOn(avx512InstructionSet))
    {
        isV512Supported = true;
        return true;
    }

    isV512Supported = false;
    return compOpportunisticallyDependsOn(InstructionSet_AVX10v1);
}
```

The key thing is to just simplify the joint checks to help keep things simpler/correct and ensure we're always correctly annotating when a given ISA is going to get used.